python/ray/train/_internal/dataset_spec.py

```diff
@@ -194,17 +193,10 @@ def get_dataset_shards(
                 )
                 dataset = dataset.window(bytes_per_window=stream_window_size).repeat()
                 # In windowed mode, we re-apply the preprocessor on each iteration.
-                if self.preprocessor or config.per_epoch_preprocessor:
-                    if self.preprocessor is not None:
-                        preprocessor = self.preprocessor
-                        if config.per_epoch_preprocessor is not None:
-                            preprocessor = Chain(
-                                preprocessor, config.per_epoch_preprocessor
-                            )
-                    else:
-                        preprocessor = config.per_epoch_preprocessor
-
-                    dataset = preprocessor._transform_pipeline(dataset)
+                if self.preprocessor is not None:
```

Comment:
Should this ever be set anymore?

Reply:
No it shouldn't be...but left all the internal logic the same for now. We would need a follow up to go and clean up all the internal code.